### DISCUSSION TITLE
PTZ presets in HA

### DISCUSSION THREAD

**Discussion by elmaswebon** [roleoroleo/yi-hack-Allwinner-v2#850] [Q&A]
First of all, thanks for your great and continuos work @roleoroleo. It's incredible everything you've achieved with these basic cameras.

I had another Yi Cams with yi-hack before, but I have just added a Dome U 2Kh 51ga SFUS, my first one with PTZ.

I have successfully added it to HA, but I can't use the PTZ presets from there.

I can move it using the onvif.ptz service:

<img width="694" alt="image" src="https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/133041914/7c96f4bd-372d-4cc3-a46b-9c05f8d63e9c">

But when trying to use the presets this way, nothing happens:

<img width="695" alt="image" src="https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/133041914/be2f2c94-1b72-4ff7-915b-e85e3ddf72fa">

I´m using the numbers or the names of the presets but nothing works.

These are the actual saved presets and they are perfectly working from ODM for example:

<img width="386" alt="image" src="https://github.com/roleoroleo/yi-hack-Allwinner-v2/assets/133041914/5fe72516-ab31-4642-87e0-908440adf9ae">

Is there any way I can call this presets from HA?

Thank you very much in advance.

**Comment by roleoroleo** [comment]
If you install my integration, you can use ptz service.
About onvif, I tried to use it to move my cam and it doesn't work.
I started a network capture session and I don't see any packet in the cam (from ha).
So, probably it doesn't work.

> I can move it using the onvif.ptz service:

IMHO, ha is not a good onvif client, or I don't understand how it works.
When you select the value of the movement, ha shows just numbers from 0 to 1.
It's not correct, it should ask to the cam what's the supported range and set the limits based on the response.
Example: yi-hack works in degrees and I set tilt from 0° to 180° and pan from 0° to 360°.
Probably ha uses percentage values, but I don't know...

**Reply by elmaswebon** [reply]
I had already your integration but didn´t notice the "Move to a preset" service.

It´s simply pefect.

Thank you very much for your work.
